src/mistral_common/tokens/tokenizers/audio.py

```diff
@@ -178,16 +190,51 @@ def _encode_audio_chunk(self, content: AudioChunk) -> AudioEncoding:
             audio=audio,
         )
 
-    def __call__(self, content: AudioChunk) -> AudioEncoding:
-        r"""Call the encoder on an audio chunk.
+    def _encode_audio_chunk(self, content: AudioChunk) -> AudioEncoding:
+        audio = Audio.from_raw_audio(content.input_audio)
+        return self._encode_audio(audio)
+
+    def _encode_audio_url_chunk(self, content: AudioURLChunk) -> AudioEncoding:
+        url = content.audio_url if isinstance(content.audio_url, str) else content.audio_url.url
+
+        try:
+            url_path = Path(url)
+            exist_path = url_path.exists()
+        except OSError:  # File name too long
+            exist_path = False
+
+        if exist_path:
+            audio = Audio.from_file(url)
+        elif urlparse(url).scheme in _URL_SCHEMES:
+            audio = Audio.from_url(url)
+        else:
+            try:
+                audio = Audio.from_base64(url)
+            except Exception:
+                raise ValueError(
+                    (
+                        "Failed to decode audio from the provided AudioURLChunk. Please ensure the URL is a local file "
+                        "path, a valid URL, or a base64-encoded audio string."
+                    )
+                )
```

Comment:
Ok for now for me!

IMO it's a bit cleaner if we could try to classify the url into one of the different types:

url_type: URLType = retrieve_url_type(...) # returns an enum

and then we do

```py
if audio_type == AudioType.URL:

elif == FILE

elif Base64

....
```